fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -0,0 +1,421 @@
+                    ]
+                    neighbors_ids.append(unique_id)
+                    break
+
+        return self.samples.select([sample_id] + neighbors_ids)
+
+    def no_leaks_view(self, view):
+        return view.exclude(self.leaks.values("id"))
+
+    def tag_leaks(self, tag="leak"):
+        """Tag leaks"""
+        for s in self.leaks.iter_samples():
+            s.tags.append(tag)
+            s.save()
+
+    def _id2splitConstructor(self):
+
+        # do this once at the beggining of the run
+        # has O(n) memory cost but memory is cheap compared to
+        # doing a couple of `in` operations per sample every run
+        # I want that sweet sweet O(1) lookup
+        id2split = {}
+        for split_name, split_view in self.split_views.items():
+            sample_ids = split_view.values("id")
+            id2split.update({sid: split_name for sid in sample_ids})
+
+        return id2split
+
+
+def _to_views(samples, split_views=None, split_field=None, split_tags=None):
+    """Helper function so that we can always work with views"""
+
+    arithmetic_true = lambda x: int(x is not None)
+    num_given = (
+        arithmetic_true(split_views)
+        + arithmetic_true(split_field)
+        + arithmetic_true(split_tags)
+    )
+
+    if num_given == 0:
+        raise ValueError(f"One of the split arguments must be given.")
+    if num_given > 1:
+        raise ValueError(f"Only one of the split arguments must be given.")
+
+    if split_views:
+        return split_views
+
+    if split_field:
+        return _field_to_views(samples, split_field)
+
+    if split_tags:
+        return _tags_to_views(samples, split_tags)
+
+
+def _field_to_views(samples, field):
+    field_values = samples.distinct(field)
+
+    if len(field_values) < 2:
+        raise ValueError(
+            f"Field {field} has less than 2 distinct values,"
+            f"can't be used to create splits"
+        )
+
+    views = {}
+    for val in field_values:
+        view = samples.match(F(field) == val)
+        views[val] = view
+
+    return views
+
+
+def _tags_to_views(samples, tags):
+    if len(tags) < 2:
+        raise ValueError("Must provide at least two tags.")
+
+    views = {}
+    for tag in tags:
+        view = samples.match_tags([tag])
+        if len(view) < 1:  # no samples in tag
+            raise ValueError(
+                f"One of the tags provided, '{tag}', has no samples. Make sure every tag has at least one sample."
+            )
+        views[tag] = view
+
+    for i, (tag, view) in enumerate(views.items()):
+        other_tags = [t for t in tags if not t == tag]
+        if len(view.match_tags(other_tags)) > 0:
+            raise ValueError(
+                f"One or more samples have more than one of the tags provided! Every sample should have at most one of the tags provided."
+            )
+
+    return views
+
+
+def _throw_index_is_bigger_warning(sample_id):
```

Comment:
I like this idea, but it gets very annoying when this happens 300 times in a row

Reply:
I now make it very explicit in the docs that the similarity's samples should be equal to the samples passed to leaky splits which should equal the union of the splits. I think if the user chooses to ignore that they should either deal with the annoyance or suppress warnings. What's the alternative for people that make the mistake unwittingly and need an error?